harness/determined/common/storage/base.py

```diff
@@ -159,3 +162,26 @@ def _list_directory(root: Union[str, os.PathLike]) -> Dict[str, int]:
                 result[rel_path] = os.path.getsize(abs_path)
 
         return result
+
+
+def from_string(shortcut: str) -> StorageManager:
+    p: urllib.parse.ParseResult = urllib.parse.urlparse(shortcut)
+    if any((p.params, p.query, p.fragment)):
+        raise ValueError(f'Malformed checkpoint_storage string "{shortcut}"')
+    scheme = p.scheme.lower()
+    if scheme in ["", "file"]:
+        base_path = p.path
+        return storage.SharedFSStorageManager(base_path=base_path)
```

Comment:
what about the `file:///localhost:1234/a/b/c` or the `localhost:1234/a/b/c` case, are we silently accepting those cases?

What happens if somebody uses too few forward slashes, like `file://a/b/c` instead of `file:///a/b/c`?